src/misc/unicode.cpp

```diff
@@ -1243,21 +1512,147 @@ static void import_mapping_ascii(const std_fs::path& path_root)
+	// Import lowercase <-> uppercase mapping data
+
+	assert(uppercase.empty());
+	assert(lowercase.empty());
+
+	// Open the file
+	const auto& file_name = file_name_case;
+	auto in_file = open_mapping_file(path_root, file_name);
+	if (!in_file) {
+		return;
+	}
+
+	// Read and parse
+	std::string line_str = "";
+	size_t line_num      = 0;
+
+	// All the encountered uppercase/lowercase code points
+	std::set<uint16_t> all_uppercase = {};
+	std::set<uint16_t> all_lowercase = {};
+
+	map_code_point_case_t new_uppercase = {};
+	map_code_point_case_t new_lowercase = {};
+
+	while (get_line(in_file, line_str, line_num)) {
+		std::vector<std::string> tokens;
+		if (!get_tokens(line_str, tokens)) {
+			continue; // empty line
+		}
+
+		if (tokens.size() != 2) {
+			error_parsing(file_name, line_num);
+			return;
+		}
+
+		uint16_t code_point_upper = 0;
+		uint16_t code_point_lower = 0;
+
+		// Skip code points which do not have corresponding upper/lower
+		// case characters - they would be needed if we were providing
+		// functions like is_upper_case(...)
+
+		if (tokens[0] == "NNN") {
+			// Retrieve code point
+			if (!get_hex_16bit(tokens[1], code_point_lower)) {
+					error_parsing(file_name, line_num);
+					return;
+			}
+
+			// Make sure there are no repetitions
+			if (all_lowercase.count(code_point_lower) ||
+			    all_uppercase.count(code_point_lower)) {
+					error_code_point_found_twice(code_point_lower,
+					                             file_name,
+					                             line_num);
+					return;
+			}
+
+			// Store the code point for further repetition checks
+			all_lowercase.insert(code_point_lower);
+			continue;
+		}
+
+		if (tokens[1] == "NNN") {
+			// Retrieve code point
+			if (!get_hex_16bit(tokens[0], code_point_upper)) {
+					error_parsing(file_name, line_num);
+					return;
+			}
+
+			// Make sure there are no repetitions
+			if (all_lowercase.count(code_point_upper) ||
+			    all_uppercase.count(code_point_upper)) {
+					error_code_point_found_twice(code_point_upper,
+					                             file_name,
+					                             line_num);
+					return;
+			}
+
+			// Store the code point for further repetition checks
+			all_uppercase.insert(code_point_upper);
+			continue;
+		}
+
+		// Retrieve code points
+		if (!get_hex_16bit(tokens[0], code_point_upper) ||
+		    !get_hex_16bit(tokens[1], code_point_lower)) {
+			error_parsing(file_name, line_num);
+			return;
+		}
+
+		// Make sure there are no repetitions
+		if (all_lowercase.count(code_point_lower) ||
+		    all_uppercase.count(code_point_lower)) {
+			error_code_point_found_twice(code_point_lower,
+			                             file_name,
+			                             line_num);
+			return;
+		}
+		if (all_lowercase.count(code_point_upper) ||
+		    all_uppercase.count(code_point_upper)) {
+			error_code_point_found_twice(code_point_upper,
+			                             file_name,
+			                             line_num);
```

Comment:
Suggest quantity comparisons (count > 0)

Reply:
OK, changed - everywhere in this file.